DEBUG: Establishing staging directory @ C:\Users\aaulb\AppData\Local\Temp\ay_tmp_yj6trs3t
DEBUG: Looking for matching profile for: hosts: "cinema4d" | product_types: "workfile" | product_base_types: "workfile" | task_names: "lighting" | task_types: "Lighting"
DEBUG: "workfile" not found in "product_types": ['review', 'render', 'prerender']
DEBUG: "cinema4d" not found in "hosts": ['standalonepublisher']
DEBUG: "cinema4d" not found in "hosts": ['substancedesigner']
DEBUG: "cinema4d" not found in "hosts": ['maya']
DEBUG: "cinema4d" not found in "hosts": ['traypublisher']
DEBUG: "cinema4d" not found in "hosts": ['max']
DEBUG: "cinema4d" not found in "hosts": ['unreal']
DEBUG: Profile selected: {'hosts': [], 'product_types': [], 'task_names': [], 'task_types': [], 'template_name': 'default'}
DEBUG: Product: workfileLighting
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 419
DEBUG: Response <RestApiResponse [200]>
DEBUG: Prepared product: workfileLighting
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 105
DEBUG: Response <RestApiResponse [200]>
DEBUG: Source: {root[work]}/Test_01/shots/sh0010/work/lighting/test1_sh0010_lighting_v041.c4d
DEBUG: Creating new version ...
DEBUG: Prepared version: v045
DEBUG: Executing [GET] http://ayon.styleframe.de/api/info
DEBUG: http://ayon.styleframe.de:80 "GET /api/info HTTP/1.1" 200 6829
DEBUG: Response <RestApiResponse [200]>
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 112
DEBUG: Response <RestApiResponse [200]>
DEBUG: Anatomy template name: default
DEBUG: Executing [POST] http://ayon.styleframe.de/api/projects/Test_01/operations
DEBUG: http://ayon.styleframe.de:80 "POST /api/projects/Test_01/operations HTTP/1.1" 200 221
DEBUG: Response <RestApiResponse [200]>
INFO: Product 'workfileLighting' version 45 written to database..
DEBUG: Integrating source files to destination ...
DEBUG: Checking file ... K:\work\Test_01\shots\sh0010\work\lighting\test1_sh0010_lighting_v041.c4d -> K:\work\Test_01\shots\sh0010\workfile\workfileLighting\v045\test1_sh0010_workfileLighting_v045.c4d
DEBUG: Copying file ... K:\work\Test_01\shots\sh0010\work\lighting\test1_sh0010_lighting_v041.c4d -> K:\work\Test_01\shots\sh0010\workfile\workfileLighting\v045\test1_sh0010_workfileLighting_v045.c4d
DEBUG: Backed up existing files: []
DEBUG: Transferred files: ['K:\\work\\Test_01\\shots\\sh0010\\workfile\\workfileLighting\\v045\\test1_sh0010_workfileLighting_v045.c4d']
DEBUG: Retrieving Representation Site Sync information ...
DEBUG: [{'id': '1d7fae9e-7fca-4176-8769-cc4b50e7a1bd', 'entity_type': 'representation', 'project_name': 'Test_01', 'operation': 'create', 'data': {'id': 'e67f81cafb6e11f0bed0ac50de9e90ce', 'versionId': 'e6795237fb6e11f09c37ac50de9e90ce', 'files': [{'id': 'e69d3883fb6e11f0ae74ac50de9e90ce', 'name': 'test1_sh0010_workfileLighting_v045.c4d', 'path': '{root[work]}/Test_01/shots/sh0010/workfile/workfileLighting/v045/test1_sh0010_workfileLighting_v045.c4d', 'size': 144284, 'hash': 'test1_sh0010_workfileLighting_v045,c4d|1769509376,6512353|144284', 'hash_type': 'op3'}], 'name': 'c4d', 'data': {'context': {'root': {'work': 'K:/work'}, 'project': {'name': 'Test_01', 'code': 'test1'}, 'hierarchy': 'shots', 'folder': {'name': 'sh0010', 'type': 'Shot', 'path': '/shots/sh0010', 'parents': ['shots']}, 'product': {'name': 'workfileLighting', 'type': 'workfile'}, 'version': 45, 'ext': 'c4d', 'task': {'name': 'lighting', 'type': 'Lighting', 'short': 'lgt'}, 'representation': 'c4d', 'username': 'SF_Ayon_Admin', 'user': {'name': 'SF_Ayon_Admin'}}}, 'attrib': {'path': 'K:\\work\\Test_01\\shots\\sh0010\\workfile\\workfileLighting\\v045\\test1_sh0010_workfileLighting_v045.c4d', 'template': '{root[work]}/{project[name]}/{hierarchy}/{folder[name]}/{product[type]}/{product[name]}/v{version:0>3}/{project[code]}_{folder[name]}_{product[name]}_v{version:0>3}<_{output}><.{frame:0>4}><_{udim}>.{ext}'}}}]
DEBUG: Executing [POST] http://ayon.styleframe.de/api/projects/Test_01/operations
DEBUG: http://ayon.styleframe.de:80 "POST /api/projects/Test_01/operations HTTP/1.1" 200 228
DEBUG: Response <RestApiResponse [200]>
INFO: Registered 1 representations: c4d
DEBUG: Establishing staging directory @ C:\Users\aaulb\AppData\Local\Temp\ay_tmp_vffh_f4f
DEBUG: Looking for matching profile for: hosts: "cinema4d" | product_types: "review" | product_base_types: "review" | task_names: "lighting" | task_types: "Lighting"
DEBUG: "cinema4d" not found in "hosts": ['standalonepublisher']
DEBUG: "cinema4d" not found in "hosts": ['substancedesigner']
DEBUG: "cinema4d" not found in "hosts": ['maya']
DEBUG: "cinema4d" not found in "hosts": ['traypublisher']
DEBUG: "cinema4d" not found in "hosts": ['max']
DEBUG: "cinema4d" not found in "hosts": ['unreal']
DEBUG: Profile selected: {'hosts': [], 'product_types': ['review', 'render', 'prerender'], 'task_names': [], 'task_types': [], 'template_name': 'render'}
DEBUG: Product: reviewMain
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 409
DEBUG: Response <RestApiResponse [200]>
DEBUG: Prepared product: reviewMain
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 105
DEBUG: Response <RestApiResponse [200]>
DEBUG: Source: {root[work]}/Test_01/shots/sh0010/work/lighting/test1_sh0010_lighting_v041.c4d
DEBUG: Creating new version ...
DEBUG: Prepared version: v029
DEBUG: http://ayon.styleframe.de:80 "POST /graphql HTTP/1.1" 200 112
DEBUG: Response <RestApiResponse [200]>
DEBUG: Anatomy template name: render
DEBUG: Template filled: K:/render/Test_01/shots/sh0010/publish/review/reviewMain/v029/test1_sh0010_reviewMain_v029.0996.jpg
DEBUG: Anatomy template name: render
DEBUG: Anatomy template name: render
DEBUG: Template filled: K:/render/Test_01/shots/sh0010/publish/review/reviewMain/v029/test1_sh0010_reviewMain_v029.0996.jpg
Traceback (most recent call last):
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\plugins\publish\integrate.py", line 154, in process
    self.register(instance, file_transactions, filtered_repres)
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\plugins\publish\integrate.py", line 247, in register
    file_transactions.add(src, dst)
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\lib\file_transaction.py", line 95, in add
    raise DuplicateDestinationError(
ayon_core.lib.file_transaction.DuplicateDestinationError: Transfer to destination is already in queue: C:\Users\aaulb\AppData\Local\Temp\ay_tmp_vffh_f4f\A_reviewMain0996.jpg -> K:\render\Test_01\shots\sh0010\publish\review\reviewMain\v029\test1_sh0010_reviewMain_v029.0996.jpg. It's not allowed to be replaced by a new transfer from C:\Users\aaulb\AppData\Local\Temp\ay_tmp_vffh_f4f\reviewMain0996.jpg

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\dependency_packages\ayon_2510091624_windows.zip\dependencies\pyblish\plugin.py", line 528, in __explicit_process
    runner(*args)
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\plugins\publish\integrate.py", line 159, in process
    raise KnownPublishError(exc).with_traceback(sys.exc_info()[2])
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\plugins\publish\integrate.py", line 154, in process
    self.register(instance, file_transactions, filtered_repres)
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\plugins\publish\integrate.py", line 247, in register
    file_transactions.add(src, dst)
  File "C:\Users\aaulb\AppData\Local\Ynput\AYON\addons\core_1.7.2\ayon_core\lib\file_transaction.py", line 95, in add
    raise DuplicateDestinationError(
ayon_core.pipeline.publish.publish_plugins.KnownPublishError: Transfer to destination is already in queue: C:\Users\aaulb\AppData\Local\Temp\ay_tmp_vffh_f4f\A_reviewMain0996.jpg -> K:\render\Test_01\shots\sh0010\publish\review\reviewMain\v029\test1_sh0010_reviewMain_v029.0996.jpg. It's not allowed to be replaced by a new transfer from C:\Users\aaulb\AppData\Local\Temp\ay_tmp_vffh_f4f\reviewMain0996.jpg

--- a/client/ayon_cinema4d/api/lib.py
+++ b/client/ayon_cinema4d/api/lib.py
@@ -650,3 +650,38 @@ def generate_review(files, output_path, fps=24):
             os.remove(list_file_path)
 
     return output_path
+
+
+def generate_thumbnail(source_file, output_path):
+    """
+    Generate a thumbnail (JPG) from a source image file using ffmpeg.
+
+    Args:
+        source_file (str): Path to source image.
+        output_path (str): Path to output JPG.
+    """
+    log = logging.getLogger(__name__)
+
+    ffmpeg_cmd = get_ffmpeg_tool_args("ffmpeg")
+
+    cmd = ffmpeg_cmd + [
+        "-y",
+        "-i", source_file,
+        "-vframes", "1",
+        output_path
+    ]
+
+    log.info(f"Generating thumbnail: {' '.join(cmd)}")
+
+    try:
+        subprocess.run(
+            cmd,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+    except subprocess.CalledProcessError as e:
+        log.error(f"FFmpeg thumbnail generation failed: {e.stderr.decode()}")
+        raise
+
+    return output_path

--- a/client/ayon_cinema4d/plugins/publish/extract_redshift_render.py
+++ b/client/ayon_cinema4d/plugins/publish/extract_redshift_render.py
@@ -178,17 +178,23 @@ class ExtractRedshiftRender(publish.Extractor):
             # Identify extension
             ext = os.path.splitext(seq_files[0])[1].lstrip(".").lower()
 
+            # Check for Alpha sequence
+            is_alpha = os.path.basename(seq_files[0]).lower().startswith("a_")
+
             repre = {
-                "name": ext,
+                "name": "alpha" if is_alpha else ext,
                 "ext": ext,
                 "files": seq_files if len(seq_files) > 1 else seq_files[0],
                 "stagingDir": staging_dir,
             }
 
+            if is_alpha:
+                repre["outputName"] = "alpha"
+
             instance.data.setdefault("representations", []).append(repre)
 
             # Generate review from sequence
-            if ext not in ["mp4", "mov"] and len(seq_files) > 1:
+            if ext not in ["mp4", "mov"] and len(seq_files) > 1 and not is_alpha:
                 review_filename = f"{filename_base}_review.mp4"
                 review_path = os.path.join(staging_dir, review_filename)
                 try:

--- a/client/ayon_cinema4d/plugins/publish/extract_review.py
+++ b/client/ayon_cinema4d/plugins/publish/extract_review.py
@@ -84,6 +84,32 @@ class Cinema4DExtractReview(publish.Extractor):
 
             instance.data.setdefault("representations", []).append(representation)
 
+            # Generate thumbnail if not alpha and sequence
+            if not is_alpha and len(seq_files) > 0:
+                thumb_filename = "thumbnail.jpg"
+                thumb_path = os.path.join(dir_path, thumb_filename)
+
+                # Check if thumbnail already exists (e.g. from previous loop)
+                if not os.path.exists(thumb_path):
+                    # Pick middle frame
+                    middle_index = len(seq_files) // 2
+                    source_file = seq_files[middle_index]
+                    source_path = os.path.join(dir_path, source_file)
+
+                    try:
+                        lib.generate_thumbnail(source_path, thumb_path)
+
+                        thumb_repre = {
+                            "name": "thumbnail",
+                            "ext": "jpg",
+                            "files": thumb_filename,
+                            "stagingDir": dir_path,
+                            "tags": ["thumbnail"]
+                        }
+                        instance.data.setdefault("representations", []).append(thumb_repre)
+                    except Exception as e:
+                        self.log.warning(f"Failed to generate thumbnail: {e}")
+
             # If it is an image sequence, we want to generate a review MP4
             # Skip alpha sequences for review generation
             if ext not in ["mp4", "mov"] and len(seq_files) > 1 and not is_alpha:

--- a/client/ayon_cinema4d/plugins/publish/extract_review.py
+++ b/client/ayon_cinema4d/plugins/publish/extract_review.py
@@ -64,22 +64,29 @@ class Cinema4DExtractReview(publish.Extractor):
             # Identify extension
             ext = os.path.splitext(seq_files[0])[1].lstrip(".").lower()
 
+            # Check for Alpha sequence
+            is_alpha = os.path.basename(seq_files[0]).lower().startswith("a_")
+
             # Main representation (sequence or movie)
             representation = {
-                "name": ext,
+                "name": "alpha" if is_alpha else ext,
                 "ext": ext,
                 "files": seq_files if len(seq_files) > 1 else seq_files[0],
                 "stagingDir": dir_path,
             }
 
+            if is_alpha:
+                representation["outputName"] = "alpha"
+
             # If it is a video file, tag it as review
-            if ext in ["mp4", "mov"]:
+            if ext in ["mp4", "mov"] and not is_alpha:
                 representation["tags"] = ["review", "preview", "ftrackreview"]
 
             instance.data.setdefault("representations", []).append(representation)
 
             # If it is an image sequence, we want to generate a review MP4
-            if ext not in ["mp4", "mov"] and len(seq_files) > 1:
+            # Skip alpha sequences for review generation
+            if ext not in ["mp4", "mov"] and len(seq_files) > 1 and not is_alpha:
                 # Generate review
                 review_filename = f"{filename}.mp4"
                 review_path = os.path.join(dir_path, review_filename)

--- a/client/ayon_cinema4d/plugins/publish/extract_review.py
+++ b/client/ayon_cinema4d/plugins/publish/extract_review.py
@@ -80,7 +80,7 @@ class Cinema4DExtractReview(publish.Extractor):
 
             # If it is a video file, tag it as review
             if ext in ["mp4", "mov"] and not is_alpha:
-                representation["tags"] = ["review", "preview", "ftrackreview"]
+                representation["tags"] = ["review", "ftrackreview"]
 
             instance.data.setdefault("representations", []).append(representation)
 
@@ -128,7 +128,7 @@ class Cinema4DExtractReview(publish.Extractor):
                         "ext": "mp4",
                         "files": review_filename,
                         "stagingDir": dir_path,
-                        "tags": ["review", "preview", "ftrackreview"]
+                        "tags": ["review", "ftrackreview"]
                     }
                     instance.data["representations"].append(review_repre)
                     self.log.info(f"Generated review mp4: {review_path}")

